craft_application/application.py

```diff
@@ -332,6 +344,9 @@ def _resolve_project_path(self, project_dir: pathlib.Path | None) -> pathlib.Pat
         directory. Applications may wish to override this if the project file could be
          in multiple places within the project directory.
         """
+        warnings.warn(
+            DeprecationWarning("NEEDS REMOVING BEFORE MERGE TO MAIN"), stacklevel=1
```

Comment:
is this being removed on a later task?

Reply:
Thanks for catching this! At first I thought it would need to stay until a future task, but it looks like I have indeed removed all usages.